src/app/game/game-form/game-form.component.ts

```diff
@@ -1,24 +1,65 @@
-import { Component, HostBinding } from "@angular/core";
+import { Component, HostBinding, OnDestroy, OnInit } from "@angular/core";
 import { FormControl, Validators } from "@angular/forms";
 import { GameService } from "../game.service";
+import { ActivatedRoute, Router } from "@angular/router";
+import { Subscription, filter, map } from "rxjs";
+import { GameT } from "../game.types";
+
+type ModeT = "Adicionar" | "Atualizar";
 
 @Component({
   selector: "app-game-form",
   templateUrl: "./game-form.component.html",
 })
-export class GameFormComponent {
+export class GameFormComponent implements OnInit, OnDestroy {
   @HostBinding("class") classes = "flex flex-col gap-2";
 
   name = new FormControl("", [Validators.required, Validators.minLength(3)]);
```

Comment:
recomendo tentar colocar uma tipagem aqui para que o compilador do typescript saiba que FormControl é uma string, `name = new FormControl<string>(...`, assim não seria mais necessário ajustes manuais como: `this.name.value as string`